From ba9b334598157991fdb672163a4793ef273df9f1 Mon Sep 17 00:00:00 2001
From: Frederick Mayle <fmayle@google.com>
Date: Wed, 13 Mar 2024 17:51:03 -0700
Subject: [PATCH 5/5] vmm_vhost: choose new ids for the non-standard messages
 types
SPDX-FileCopyrightText: 2019-2021 Alibaba Cloud. All rights reserved.
SPDX-FileCopyrightText: The Chromium OS Authors
SPDX-License-Identifier: Apache-2.0

Backend message ID 42 has been taken by VHOST_USER_SET_DEVICE_STATE_FD
and frontend message ID 6 has been taken by
VHOST_USER_BACKEND_SHARED_OBJECT_ADD.

Using large numbers to avoid future conflicts.

The only known user of the sleep/wake/snapshot/restore message types
outside of the crosvm repo is in AOSP and it can be updated as part of
the next crosvm merge. Usage of the other's is less clear.

BUG=b:301269927

Change-Id: I4ba272563117b65a3dd9aa4e936f164e863cafbc
Reviewed-on: https://chromium-review.googlesource.com/c/crosvm/crosvm/+/5369109
Reviewed-by: Daniel Verkamp <dverkamp@chromium.org>
Reviewed-by: Elie Kheirallah <khei@google.com>
Commit-Queue: Frederick Mayle <fmayle@google.com>
(cherry picked from commit e428c4ba88a26695f63929c680b4101a9b8d5cbc)
---
 vhost/src/vhost_user/message.rs | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/vhost/src/vhost_user/message.rs b/vhost/src/vhost_user/message.rs
index 9950c2f..2f58e2a 100644
--- a/vhost/src/vhost_user/message.rs
+++ b/vhost/src/vhost_user/message.rs
@@ -178,14 +178,14 @@ enum_value! {
         /// Query the backend for its device status as defined in the VIRTIO
         /// specification.
         GET_STATUS = 40,
-        /// Get a list of the device's shared memory regions.
-        GET_SHARED_MEMORY_REGIONS = 41,
         /// Begin transfer of internal state to/from the backend for migration
         /// purposes.
         SET_DEVICE_STATE_FD = 42,
         /// After transferring state, check the backend for any errors that may have
         /// occurred during the transfer
         CHECK_DEVICE_STATE = 43,
+        /// Get a list of the device's shared memory regions.
+        GET_SHARED_MEMORY_REGIONS = 1004,
     }
 }
 
@@ -205,18 +205,20 @@ enum_value! {
         VRING_CALL = 4,
         /// Indicate that an error occurred on the specific vring.
         VRING_ERR = 5,
+
+        // Non-standard message types.
         /// Indicates a request to map a fd into a shared memory region.
-        SHMEM_MAP = 6,
+        SHMEM_MAP = 1000,
         /// Indicates a request to unmap part of a shared memory region.
-        SHMEM_UNMAP = 7,
+        SHMEM_UNMAP = 1001,
         /// Virtio-fs draft: map file content into the window.
-        FS_MAP = 8,
+        FS_MAP = 1002,
         /// Virtio-fs draft: unmap file content from the window.
-        FS_UNMAP = 9,
+        FS_UNMAP = 1003,
         /// Virtio-fs draft: sync file content.
-        FS_SYNC = 10,
+        FS_SYNC = 1004,
         /// Virtio-fs draft: perform a read/write from an fd directly to GPA.
-        FS_IO = 11,
+        FS_IO = 1005,
     }
 }
 
-- 
2.44.0

